provider/core/src/response.rs

```diff
@@ -318,10 +335,15 @@ where
             <M::Yokeable as Yokeable<'a>>::Output,
             PhantomData<&'a ()>,
         ) -> <M2::Yokeable as Yokeable<'a>>::Output,
+        M::Yokeable: Clone,
     {
-        DataPayload {
-            yoke: self.yoke.map_project(f),
-        }
+        DataPayload(DataPayloadInner::Yoke(
+            match self.0 {
+                DataPayloadInner::Yoke(yoke) => yoke,
+                DataPayloadInner::Ref(r) => Yoke::new_owned(r.clone()),
```

Comment:
I changed it, but I'm not sure that's the right move. `Clone + 'static` might be the better bound, because such a clone is guaranteed to not allocate (i.e. it's basically a `ZeroFrom`), and way more types are `Clone` than are `ZeroFrom`. Case in point I had to add an implementation to make it work.

Reply:
`Clone + 'static` just means the output can't have any non-static borrows; it doesn't say anything about allocation. Of course, in practice for databake data, the inner ZeroVecs and Cows will clone into a borrowed form. But, if something like const alloc lands, that invariant might not always hold. Another, more concrete, short-term problem with `Clone` is that we need to include the code for both allocating and borrowing (on a Cow-like object), because the compiler does not know ahead of time which variant is going to be there.